packages/miniflare/test/plugins/do/index.spec.ts

```diff
@@ -603,3 +603,82 @@ test("multiple workers with DO useSQLite true and undefined does not cause optio
 		});
 	});
 });
+
+const BLOCKING_DO_SCRIPT = `
+import { DurableObject } from 'cloudflare:workers';
+
+export class BlockingDO extends DurableObject {
+	locks = new Map();
+
+	async blockedOp(n, lock) {
+		await new Promise((resolve) => {
+			this.locks.set(lock, () => resolve(lock));
+		});
+		return n + 2;
+	}
```

Comment:
NIT - could just be:


```suggestion
	blockedOp(n, lock) {
		return new Promise((resolve) => {
			this.locks.set(lock, () => resolve(lock));
		}).then(() =>  n + 2);
	}
```